ethcore/builtin/src/lib.rs

```diff
@@ -44,6 +46,22 @@ trait Pricer: Send + Sync {
 	fn cost(&self, input: &[u8], at: u64) -> U256;
 }
 
+/// Pricing for the Blake2 compression function (aka "F").
+/// Computes the price as a fixed cost per round where the number of rounds is part of the input
+/// byte slice.
+pub type Blake2FPricer = u64;
+
+impl Pricer for Blake2FPricer {
+	fn cost(&self, input: &[u8], _at: u64) -> U256 {
+		use std::convert::TryInto;
+		let (rounds_bytes, _) = input.split_at(std::mem::size_of::<u32>());
+		// Returning zero if the conversion fails is fine because `execute()` will check the length
+		// and bail with the appropriate error.
+		let rounds = u32::from_be_bytes(rounds_bytes.try_into().unwrap_or_else(|_| [0u8; 4] ));
+		U256::from(*self * rounds as u64)
```

Comment:
can it overflow? maybe let's cast it to `u128`?

Reply:
There's no need to cast, a `u32` is 4 `u8`s. the call to `mem::size_of::<u32>` would return `4`
```suggestion
		U256::from(*self as u128 * rounds as u128)
```

Reply:
```suggestion
		U256::from(*self as u128 * rounds as u128)
```